core/src/main/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetService.java

```diff
@@ -356,6 +356,7 @@ public static long getStartingSeqNo(final RecoveryTarget recoveryTarget) {
             final long globalCheckpoint = Translog.readGlobalCheckpoint(recoveryTarget.translogLocation());
             final SeqNoStats seqNoStats = recoveryTarget.store().loadSeqNoStats(globalCheckpoint);
             if (seqNoStats.getMaxSeqNo() <= seqNoStats.getGlobalCheckpoint()) {
+                assert seqNoStats.getLocalCheckpoint() <= seqNoStats.getGlobalCheckpoint();
```

Comment:
We can not make this change. We have the guarantee that the local checkpoint in the commit is bounded above by the maximum sequence number in the commit. We do not have a relationship between the local checkpoint in the commit and the global checkpoint in the translog (the local knowledge of the global checkpoint can lag behind). It's only *after* we know that the maximum sequence number is bounded above by the global checkpoint that we get the implication that the local checkpoint is bounded above by the global checkpoint and can therefore make this assertion.